packages/focusmanager/src/react/use-focus-manager.ts

```diff
@@ -13,35 +14,43 @@ const FOCUSABLE_SELECTORS = [
   'video[controls]'
 ]
 
-const defaultOptions = {
-  autofocus: true,
-  returnFocus: true,
-  trapped: true
-}
-
-export default function useFocusManager(ref, _options) {
-  const opts = { ...defaultOptions, ..._options }
-
-  const prevActiveElRef = React.useRef()
+export default function useFocusManager(
+  ref: React.RefObject<HTMLDivElement>,
+  {
+    autofocus,
+    returnFocus,
+    trapped
+  }: {
+    autofocus: boolean
+    returnFocus: boolean
+    trapped: boolean
+  }
+) {
+  const prevActiveElRef = React.useRef<HTMLElement>(null)
   React.useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     prevActiveElRef.current = document.activeElement
   }, [])
 
-  const [node, setNode] = React.useState(null)
+  const [node, setNode] = React.useState(
+    (null as unknown) as HTMLDivElement | null
+  )
```

Comment:
Instead of casting, would it work to type useState?

```suggestion
  const [node, setNode] = React.useState<HTMLDivElement>(
    null
  )
```

Reply:
This throws a type error
```
Argument of type 'null' is not assignable to parameter of type 'HTMLDivElement | (() => HTMLDivElement)'.
```

Reply:
`useState<HTMLDivElement | null>`?

I'm just attempting to use the typings on the generic function first as opposed to the casting.

Reply:
Ahh I'll try that